### PR TITLE
Refactoring and breaking apart schema related parsing into smaller modular pieces.

### PR DESCRIPTION
**Pull request by maniarathi**
Added some docstrings about function definitions, did some renaming, breaking apart some functions, and consolidating single use functions.

### CODE REVIEW

**Comment by maniarathi** [comment]
General request: please do not merge into master until @kislyuk has finished reviewing. Thanks!

File: ingest/template/schema_parser.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+            if kwargs.get('property_name') in self._identifiable:
+                property_metadata["identifiable"] = True
+
+        if 'key' in kwargs and "object" != property_metadata["value_type"]:
+            self.schema_template.append_column_to_tab(kwargs.get('key'))
+
+        if schema:
+            property_metadata["schema"] = schema
+
+        # put the user friendly to key in the lookup table
```
Comment (kislyuk) [inline ingest/template/schema_parser.py]: Please clarify or remove this comment.

File: ingest/template/schema_parser.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    def __initialise_template(self, data):
+
+        self.get_required_properties_from_metadata_schema(data)
+
+        property = self._extract_property(data)
+        if not property.schema or "type" not in property.schema.high_level_entity:
+            raise RootSchemaException(
+                "Schema must start with a root submittable type schema")
+        else:
+            # as this is top level add a retrievable property for
```
Comment (kislyuk) [inline ingest/template/schema_parser.py]: Please clarify or remove this comment.

File: ingest/template/schema_parser.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+            migration_info["target_version"] = property_migration["effective_from_target"]
+
+        migration_info = {migrated_property.split(".")[-1]: migration_info}
+        for part in reversed(migrated_property.split(".")[:-1]):
+            migration_info = {part: migration_info}
+
+        self.schema_template.put_migration(migration_info)
+        return self.schema_template
+
+    def __initialise_template(self, data):
```
Comment (kislyuk) [inline ingest/template/schema_parser.py]: Can these method names be refactored to not start with two underscores? The appropriate naming convention is to start with one underscore; two underscores are reserved for language-internal or binding-internal methods.

File: ingest/template/schema_parser.py (hunk 1 above)
Comment (maniarathi) [reply ingest/template/schema_parser.py]: Done.

File: ingest/template/schema_parser.py (hunk 2 above)
Comment (maniarathi) [reply ingest/template/schema_parser.py]: Done.

File: ingest/template/schema_parser.py (hunk 3 above)
Comment (maniarathi) [reply ingest/template/schema_parser.py]: Removed all the underscores from now until I can figure out what exactly needs to be private/public.

**Pull request merged**